ggml/src/ggml-cuda/ggml-cuda.cu

```diff
@@ -2035,97 +2056,75 @@ static void ggml_cuda_mul_mat_id(ggml_backend_cuda_context & ctx, ggml_tensor *
     dst_row.nb[2] = nb1;
     dst_row.nb[3] = nb1;
 
-    if (ne12 == 1) {
```

Comment:
I removed the `ne12 == 1` code path because that can now be handled above. The following, seemingly large diff shown on Github is due to me changing the indentation of the else branch.